components/plugin_marketplace/marketplace_list/marketplace_list.js

```diff
@@ -0,0 +1,145 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import {FormattedMessage} from 'react-intl';
+
+import {changeOpacity, makeStyleFromTheme} from 'mattermost-redux/utils/theme_utils';
+
+import MarketplaceItem from '../marketplace_item';
+
+const PLUGINS_PER_PAGE = 15;
+
+export default class MarketplaceList extends React.PureComponent {
+    static propTypes = {
+        plugins: PropTypes.array.isRequired,
+        theme: PropTypes.object.isRequired,
+    };
+
+    static getDerivedStateFromProps(props, state) {
+        if (state.page > 0 && props.plugins.length < state.pluginsPerPage) {
+            return {page: 0};
+        }
+
+        return null;
+    }
+
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            page: 0,
+            pluginsPerPage: PLUGINS_PER_PAGE,
```

Comment:
Why add this constant to the local state?

Reply:
You're right, it was a bad idea. I removed it from the local state :+1: